cdcsdk-testing/src/test/java/com/yugabyte/cdcsdk/testing/KinesisConsumerIT.java

```diff
@@ -0,0 +1,177 @@
+        cdcsdkContainer.start();
+
+        assertTrue(cdcsdkContainer.isRunning());
+
+    }
+
+    @AfterEach
+    public void dropTable() throws Exception {
+        ybHelper.execute(UtilStrings.getDropTableStmt(DEFAULT_TABLE_NAME));
+    }
+
+    @Test
+    public void automationOfKinesisAssertions() throws Exception {
+
+        int recordsInserted = 1;
+        String uuid = getUUID().substring(0, 8);
+        for (int i = 0; i < recordsInserted; ++i) {
+            ybHelper.execute(UtilStrings.getInsertStmt(DEFAULT_TABLE_NAME, i, uuid, "last_" + i, 23.45));
+        }
+        ybHelper.execute(UtilStrings.getUpdateStmt(DEFAULT_TABLE_NAME, 0, 25));
+
+        ybHelper.execute(UtilStrings.getDeleteStmt(DEFAULT_TABLE_NAME, 0));
+
+        Awaitility.await().atMost(Duration.ofSeconds(10));
+
+        List<String> expected_data = List.of(
+                "{\"id\":0,\"first_name\":\"" + uuid
+                        + "\",\"last_name\":\"last_0\",\"days_worked\":23.45,\"__deleted\":\"false\"}",
+                "{\"id\":0,\"days_worked\":25.0,\"__deleted\":\"false\"}",
+                "{\"id\":0,\"__deleted\":\"true\"}");
+
+        List<String> actual_data = new ArrayList<>();
+
+        AWSCredentials awsCredentials = new BasicSessionCredentials(System.getenv("AWS_ACCESS_KEY_ID"),
+                System.getenv("AWS_SECRET_ACCESS_KEY"),
+                System.getenv("AWS_SESSION_TOKEN"));
+
+        AmazonKinesisClient client = new AmazonKinesisClient(awsCredentials);
+
+        client.setEndpoint("https://kinesis.ap-south-1.amazonaws.com/");
+
+        final int INTERVAL = 2000;
+        ;
+
+        List<Shard> initialShardData = client.describeStream(stream_name).getStreamDescription().getShards();
+
+        // Getting shardIterators (at beginning sequence number) for reach shard
+        List<String> initialShardIterators = initialShardData.stream()
+                .map(s -> client.getShardIterator(new GetShardIteratorRequest()
+                        .withStreamName(stream_name)
+                        .withShardId(s.getShardId())
+                        .withStartingSequenceNumber(s.getSequenceNumberRange().getStartingSequenceNumber())
+                        .withShardIteratorType(ShardIteratorType.AFTER_SEQUENCE_NUMBER))
+                        .getShardIterator())
+                .collect(Collectors.toList());
+
+        String shardIterator = initialShardIterators.get(0);
+
+        // Continuously read data records from a shard
+        while (true) {
+            GetRecordsRequest getRecordsRequest = new GetRecordsRequest();
+            getRecordsRequest.setShardIterator(shardIterator);
+            getRecordsRequest.setLimit(25);
+
+            GetRecordsResult recordResult = client.getRecords(getRecordsRequest);
+
+            recordResult.getRecords().forEach(record -> {
+                try {
+                    String rec = new String(record.getData().array(), "UTF-8");
+                    actual_data.add(rec);
```

Comment:
Where are the records for a particular UUID filtered?

Reply:
Here we are collecting all records present in kinesis datastream. We are checking for required UUID in validateRecords()